parcels/codegenerator.py

```diff
@@ -54,18 +54,25 @@ def __getattr__(self, attr):
         if isinstance(getattr(self.obj, attr), Grid):
             return GridNode(getattr(self.obj, attr),
                             ccode="%s->%s" % (self.ccode, attr))
+        elif attr == "eval":
+            return FieldEvalCallNode(self)
```

Comment:
Why did you make this change, using an `elif` instead of the `__getitem__`? Is this why the CI breaks?

Reply:
Whoops! I didn't mean to get rid of `__getitem__`, that was part of my testing. As to the change itself, because we're looking at `field.eval`, it will call `__getattr__` on the `FieldNode` (as part of `self.visit(node.func)` in `visit_Call`, I think).

However, it did seem that `__getitem__` here is never called, and is incorrect: `FieldEvalNode` has a 3-argument constructor, so we can't call it as `FieldEvalNode(self.obj, attr)` without raising a `TypeError` anyway? Because of that, I would still advocate for removing it.